### ISSUE TITLE
Segmentation fault when newrank=true

### ISSUE THREAD

**Issue by Chiel92** [ellson/graphviz#1221]
Consider the following dot code:

```
digraph "Graph d718cbc6-5e20-4417-a41a-e1e380469dd2" {
	graph [
		newrank=true
	];
	subgraph "cluster_1" {
		1;
	}
	subgraph "cluster_2" {
		1
	}
}
```

Then running dot on this will segfault.
```
$ cat ~/Desktop/repro.txt | ./dot.exe -Tpng > after.png
Segmentation fault
````

However, the old ranking algorithm will not segfault on this. Is this a bug in the new ranking algorithm?

**Comment by emden** [comment]
Not exactly. The problem is that you have the same node in two incompatible clusters, i.e., non-nested. The old ranking works because it quietly ignores one of the nodes in one of the clusters.

**Comment by Chiel92** [comment]
All right, so giving a segfault is appropriate then? Shouldn't dot give an error message about invalid input?

**Comment by magneticnorth** [comment]
Its a bug, it should report the problem with the input graph and adjust it
before proceeding or exit gracefully.

**Comment by emden** [comment]
If the packmode attribute is set, this is actually checked and reported, but it doesn't exit with an error.
